### ISSUE TITLE
PIC support

### ISSUE THREAD

**Issue by sunfishcode** [stoklund/cretonne#188]
Cretonne's legalization of addresses uses absolute addresses, rather than PIC. For example, these two functions:
```
function: function %var_address() -> i64 native {
    gv0 = globalsym %var

ebb0:
    v0 = global_addr.i64 gv0
    return v0
}

function: function %func_address() -> i64 native {
    sig0 = () native
    fn0 = sig0 %func

ebb0:
    v0 = func_addr.i64 fn0
    return v0
}
```
are both legalized to `movabs $0x0,%rax`. PIC will require legalizing them to a GOT load, `mov 0x0(%rip),%rax` with a `R_X86_64_GOTPCREL` (or similar) relocation.

**Comment by stoklund** [comment]
I imagine we would have a global flag (in settings) setting the PIC mode. This then controls how these instructions are expanded or encoded.

**Comment by pchickey** [comment]
Should be closed by #206
